web/client/utils/__tests__/CatalogUtils-test.js

```diff
@@ -541,6 +541,68 @@ describe('Test the CatalogUtils', () => {
         expect(records.length).toBe(1);
     });
 
+    it('csw with DC references with implicit name in wms URI (RNDT / INSPIRE)', () => {
+        const records = CatalogUtils.getCatalogRecords('csw', {
+            records: [{
+                dc: {
+                    URI: [
+                        {
+                            TYPE_NAME: 'DC_1_1.URI',
+                            protocol: 'http://www.opengis.net/def/serviceType/ogc/wms',
+                            description: 'access point',
+                            value: 'http://geoserver/wms?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetMap&LAYERS=workspace:layer'
+                        }
+                    ]
+                }
+            }]
+        }, {});
+        expect(records.length).toBe(1);
+    });
+
+    it('csw with DC references with implicit name in wms URI (RNDT / INSPIRE) - download link - no title', () => {
+        const locales = {
+            catalog: {
+                notAvailable: "Not Available"
+            }
+        };
+        const records = CatalogUtils.getCatalogRecords('csw', {
+            records: [{
+                dc: {
+                    URI: [
+                        {
+                            protocol: 'https://registry.geodati.gov.it/metadata-codelist/ProtocolValue/www-download',
+                            description: 'access point',
+                            value: 'http://gisdata.provider.host/shp/test_layer.zip'
+                        }
+                    ]
+                }
+            }]
+        }, {}, locales);
+        const resourceLink = '<ul><li><a target="_blank" href="http://gisdata.provider.host/shp/test_layer.zip">Not Available - shapefile</a></li></ul>';
+        expect(records.length).toBe(1);
+        expect(records[0].metadata.uri.length).toBe(1);
+        expect(records[0].metadata.uri[0]).toBe(resourceLink);
+    });
+
+    it('csw with DC references with implicit name in wms URI (RNDT / INSPIRE) - download link - value only', () => {
+        const records = CatalogUtils.getCatalogRecords('csw', {
+            records: [{
+                dc: {
+                    title: 'Test Layer',
+                    URI: [
+                        {
+                            value: 'http://gisdata.provider.host/geoserver/wms?tiled=true&version=1.1.1'
```

Comment:
here you need the protocol

Reply:
I am happy to put it, but this is designed to specifically test the case where the `<dc:URI>` entry comes without any attribute. One example is the layer 'Sottopassaggi' in the C027 project datasets. While backporting I noticed the issue and provided a test for this case.